python/huggingfaceserver/huggingfaceserver/__main__.py

```diff
@@ -126,6 +127,14 @@ def list_of_strings(arg):
         choices=dtype_choices,
         help=f"data type to load the weights in. One of {dtype_choices}. Defaults to float16 for GPU and float32 for CPU systems",
     )
+    # vLLM uses max-model-len as paramter to denote max tokens. Register the same for HuggingFace (if vllm not available)
+    parser.add_argument(
+        "--max-model-len",
```

Comment:
```suggestion
        "--max-model-len",
        "--max_model_len",
```

Reply:
We're just trying to have the same argument as [vLLM](https://arc.net/l/quote/ufttkhkf). Hence the hyphen `-` instead of underscore `_`

Reply:
Also the problem with adding `--max_model_len` here is, this would not work if vLLM exists. 
If we want to support `--max_model_len`, we'd need to add it separately. This ends up creating 3 CLI flags for one variable.

Reply:
I thought `--max-model-len` is parsed to `max_model_len` as well

Reply:
Yeah --max-model-len is parsed to args.max_model_len
Just that if we want to support it, it should be added outside of `if _vllm:`

Reply:
```python
parser.add_argument('--foo_bar')
parser.add_argument('--foo_bar')
args = parser.parse_args(['--foo-bar', '24', '--foo_bar', '30'])
Namespace(foo_bar='30')
```

Reply:
if vLLM exists, we import that parser and hence it won't let us add `--max-model-len` again
so the final code would end up looking like
```python
parser.add_argument('--max_length', dest='max_model_len')
parser.add_argument('--max_model_len')
if not vllm_available():
    parser.add_argument('--max-model-len')
```

All three for essentially the same variable. Do we want this?

Reply:
vllm actually implements a flexible arg parser so it can automatically convert underscore to dashes
https://github.com/vllm-project/vllm/blob/80162c44b1d1e59a2c10f65b6adb9b0407439b1f/vllm/utils.py#L1088